src/crimson/admin/osd_admin.cc

```diff
@@ -245,8 +245,11 @@ class DumpMetricsHook : public AdminSocketHook {
     case data_type::GAUGE:
       f->dump_float(value_name, v.d());
       break;
+    case data_type::REAL_COUNTER:
+      f->dump_float(value_name, v.d());
+      break;
     case data_type::COUNTER:
-      f->dump_unsigned(value_name, v.ui());
+      f->dump_int(value_name, v.i());
```

Comment:
`COUNTER` should be considered as unsigned, i think.

Reply:
Um, the most recent version of seastar seems to allow negative counter: https://github.com/scylladb/seastar/blob/693dac6c030e02cab3ab87a4c818044d935837cc/src/core/reactor.cc#L2279

Reply:
i see. seastar does not enforce the signness of this type, and it stores it using a `double`.  and it considers it as an unsigned when exporting it to collected, see https://github.com/scylladb/seastar/blob/693dac6c030e02cab3ab87a4c818044d935837cc/src/core/scollectd.cc#L217-L220

but considers as a signed when exporting it to prometheus, see https://github.com/scylladb/seastar/blob/693dac6c030e02cab3ab87a4c818044d935837cc/src/core/prometheus.cc#L63-L64 . but according to prometheus' doc (https://prometheus.io/docs/concepts/metric_types/), a counter can only be unsigned. so i think unsigned is a better choice. also, an unsigned integer is able to represent a greater value.

Reply:
> i see. seastar does not enforce the signness of this type, and it stores it using a `double`. and it considers it as an unsigned when exporting it to collected, see https://github.com/scylladb/seastar/blob/693dac6c030e02cab3ab87a4c818044d935837cc/src/core/scollectd.cc#L217-L220
> 
> but considers as a signed when exporting it to prometheus, see https://github.com/scylladb/seastar/blob/693dac6c030e02cab3ab87a4c818044d935837cc/src/core/prometheus.cc#L63-L64 . but according to prometheus' doc (https://prometheus.io/docs/concepts/metric_types/), a counter can only be unsigned. so i think unsigned is a better choice. also, an unsigned integer is able to represent a greater value.

Yes, I agree. But seastar's metric `cpu_steal_time_ms` being negative is causing admin `dump_metrics` command's failure, right now. Should we drop a patch to the seastar project?

Reply:
i'd do so if i have had an explanation and a sound solution to this issue. to my best understanding, `reactor::total_steal_time()` should return a duration in nanoseconds. by reading the comment of that member function, i think this value should be positive. 

do you have a fix? if yes, would be better to upstream it.

Reply:
Um, not yet. I actually don't quite understand why that value can be negative......